lib/configuration.js

```diff
@@ -208,11 +208,25 @@ Configuration.prototype._checkLocalServiceContext = function() {
   //   the service name.  If neither of these conditions are true, the 
   //   value of the GAE_MODULE_NAME environment variable should be used as the 
   //   service name.
+  //
+  // To determine the service version to use:
+  //   If the user species a version, then that version will be used.
+  //   Otherwise, the value of the environment variable GAE_MODULE_VERSION
+  //   will be used if and only if the FUNCTION_NAME environment variable is
+  //   not set.
   var funcName = env.FUNCTION_NAME;
-  var service = isUndefined(funcName) ? env.GAE_MODULE_NAME : funcName;
-  var gaeModuleVer = env.GAE_MODULE_VERSION;
+  var service;
+  var version;
+  if (isUndefined(funcName)) {
+    service = env.GAE_MODULE_NAME;
+    version = env.GAE_MODULE_VERSION;
+  } else {
+    service = funcName;
+  }
+
   this._serviceContext.service = isString(service) ? service : '';
-  this._serviceContext.version = isString(gaeModuleVer) ? gaeModuleVer : '';
+  this._serviceContext.version = isString(version) ? version : '';
```

Comment:
I've updated the version to default to being undefined instead of the empty string.